README.md

```diff
@@ -1,28 +1,123 @@
+CRL's are created in a way that is only valid at the point of the certificate issuer. 
+OCSP is not easily prefetchable without a cooperating client implementing OCSP stapling. 
+Both CRL's and OCSP bring significant risk to the uptime of a network. 
+This denylist has been implemented in such a way that downtime is tolerated and the entire contents are prefetched and cacheable.
 
-## Updating the denylist
+## Supported key types
+
+The denylist currently only supports PEM encoded ed25519 keys.
+
+# Instructions for adding a certificate to the denylist
+
+## Get certificate to block
+The first step is to find the certificate to block.
+Identify the peer whose certificate needs to be blocked and get the certificate from the peer diagnostics.
+Make the following API call to a nuts-node to get the peer diagnostics.
+
+```
+GET /internal/network/v1/diagnostics/peers
+```
+
+Lookup the peer by its `peerID` and copy the certificate.
+Store this PEM encoded certificate in its own file in the `certs/` directory.
+
+## Add certificate to denylist
+The unsigned denylist is maintained at `config/certs.json`.
+
+Make a new entry using
+```shell
+# generate a denylist entry
+go run cli/dumpcert/main.go --cert <path-to-cert>
+```
+
+This will produce something like:
+```json
+{
+    "issuer": "CN=Nuts Development Network Root CA",
+    "serialnumber": "86909351591313461694061157857735660045310273765",
+    "jwkthumbprint": "xaVihytdhiID5wqCOsQbbb2_GuIb0r7m1GUgPIdBDjo",
+    "reason": "TODO: Give me one reason, give me just one reason why"
+}
+```
+
+Copy this into `config/certs.json` and set the reason for blocking the certificate.
+
+
+## Sign denylist
+The key to sign the denylist is stored in the `NutsDenylist` vault on 1password in the the `PrivateKey` item.
+
+### 1password CLI (preferred)
+Install v2 of the 1password CLI by following the [instructions](https://developer.1password.com/docs/cli/get-started/) if needed.
+```shell
+# sign in to the correct account
+op signin
+
+# point the env variable to the correct value in 1password
+export DENYLIST_PRIVATEKEY_PEM="op://NutsDenylist/PrivateKey/password"
+
+# sign by letting 1password substitute the key
+op run make denylist
+```
+
+### Copy key to CLI (only if absolutely needed)
+If the 1password CLI is not an option for some reason, the denylist can also be signed after manually setting the env variable.
+```
+# get the key from 1password and subsitute below to sign the new denylist.
+DENYLIST_PRIVATEKEY_PEM="<private-key>" make denylist
+```
+
+
+## Publish denylist
+The previous step updated the signed denylist at `denylist-out/denylist.jws`. (DO NOT CHANGE THIS FILE LOCATION/NAME)
+Assuming the certificate to block was added to `certs/`, validate that the new denylist actually identifies the certificate.
+```shell
+# validate that the added certifcate is on the denylist
+make test
```

Comment:
I would have called it `make sure`. Just kidding.

Reply:
must resist... urge.. to change..